server/src/main/java/io/crate/execution/engine/window/WindowFrameState.java

```diff
@@ -49,6 +49,10 @@ public int upperBoundExclusive() {
         return upperBoundExclusive;
     }
 
+    public int partitionEnd() {
```

Comment:
`WindowFrameState` implementation assumes direct access as the only usage by utilizing `getRowInFrameAtIndexOrNull` or `getRowInPartitionAtIndexOrNull`. But with `ignore nulls` option introduced, sequential accesses are required (to search for non-null elements) and need `partitionEnd` for that.

Reply:
since `lead/lag` should not be bounded by window definitions, `upperBoundExclusive()` or `size()` are not feasible.